### ISSUE TITLE
Proposal: add a chpldoc attribute to hide the implementation type of a symbol

### ISSUE THREAD

**Issue by riftEmber** [chapel-lang/chapel#22461]
While trying to [hide the underlying implementation types of `c_ptr` and `c_array` from docs](https://github.com/chapel-lang/chapel/issues/21385), I discovered there is really no easy way to do that at the moment for non-builtin types.

On that issue Brad suggested the addition of an attribute that causes a symbol to be rendered as `type c_ptr` (or whatever symbol) instead of `class c_ptr` or `record c_ptr`. (I think we might want this for other aggregate types (`union`s) as well, or even more symbols.) I am in favor of this suggestion, though we would need some design discussion on the attribute.

This is a proposal for post-2.0; I used a hacky solution for the immediate problem in CTypes.

**Comment by bradcray** [comment]
@jabraham17 / @mppf: While working on CHANGES.md, I'm realizing that `owned` and `shared` are arguably good candidates for this as well (though they may also be amenable to just nodoc-ing their type declarations and making their methods into secondary methods?)

**Comment by mppf** [comment]
Yes, it is arguably confusing that they are listed as records today. I think just no-doc'ing them and making the methods into secondary methods sounds great for those two.

**Comment by bradcray** [comment]
Comparing https://chapel-lang.org/docs/1.31/language/spec/classes.html#SharedObject.shared.adopt to https://chapel-lang.org/docs/1.31/language/spec/bytes.html#Bytes.bytes.createBorrowingBuffer (say), I'm inclined to agree, and wondering more generally whether we might want to prefer defining methods that are chpldoc'd as secondary methods for the sake of clearer documentation (in that there's no need to scroll to see whether a procedure is a method (and if so of what type) vs. standalone).

**Comment by lydia-duncan** [comment]
https://github.com/chapel-lang/chapel/issues/5953 would also help with that

**Comment by mppf** [comment]
If we think it's better for documentation all the time for them to appear as secondary methods, we can just make chpldoc output them that way in the docs whether they are declared primary or secondary.

**Comment by arifthpe** [comment]
Attribute implemented at https://github.com/chapel-lang/chapel/pull/27432

**Comment by bradcray** [comment]
Cool!
